core/src/test/scala/matryoshka/instances/fixedpoint/stream.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2014–2016 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package matryoshka.instances.fixedpoint
+
+import matryoshka._
+import matryoshka.patterns._
+
+import scala.Int
+
+import org.specs2.ScalaCheck
+import org.specs2.mutable._
+import scalaz._, Scalaz._
+
+class StreamSpec extends Specification with ScalaCheck with specs2.scalaz.Matchers {
+
+  /** Continues the Fibonacci sequence from a seed pair of values.
+    */
+  // FIXME: Why does this drop the initial 1?
+  val fib: Coalgebra[(Int, ?), (Int, Int)] = {
+    case (p2, p1) =>
+      val c = p2 + p1
+      (c, (p1, c))
+  }
+
+  /** Generates an infinite stream of the carrier value.
+    */
+  def constantly[A]: Coalgebra[(A, ?), A] = i => (i, i)
+
+  "fib" should {
+    "begin with 1" in {
+      (0, 1).ana[Nu, (Int, ?)](fib).head must equal(1)
```

Comment:
Yeah, I like that approach. I might call `iter2` something like `binarySequence`(?) and move it out of tests.

And yeah, I get why the sequence doesn’t start where I expect it to. I guess I could seed with `(1, 0)`, but it would be nice to include the seed in the final sequence …

Maybe call it `binarySequenceFrom`?